front/components/vaults/VaultResourcesList.tsx

```diff
@@ -120,30 +124,45 @@ export const VaultResourcesList = ({
   plan,
   isAdmin,
   vault,
+  systemVault,
   category,
   onSelect,
 }: VaultResourcesListProps) => {
   const router = useRouter();
+  const sendNotification = React.useContext(SendNotificationsContext);
+
   const [showDatasourceLimitPopup, setShowDatasourceLimitPopup] =
     useState(false);
   const [showAddFolderModal, setShowAddFolderModal] = useState(false);
   const [showAddWebsiteModal, setShowAddWebsiteModal] = useState(false);
 
+  const [showDataSourcesModal, setShowDataSourcesModal] = useState(false);
   const [dataSourceSearch, setDataSourceSearch] = useState<string>("");
 
   const { dataSources, isDataSourcesLoading } = useDataSources(owner);
 
-  const managedDataSources = dataSources.filter(
-    (ds) => ds.connectorProvider && ds.connectorProvider !== "webcrawler"
-  );
   const searchBarRef = useRef<HTMLInputElement>(null);
 
-  const { vaultDataSourceViews, isVaultDataSourceViewsLoading } =
-    useVaultDataSourceViews({
-      workspaceId: owner.sId,
-      vaultId: vault.sId,
-      category: category,
-    });
+  // DataSources Views of the current vault.
+  const {
+    vaultDataSourceViews,
+    isVaultDataSourceViewsLoading,
+    mutateVaultDataSourceViews,
+  } = useVaultDataSourceViews({
+    workspaceId: owner.sId,
+    vaultId: vault.sId,
+    category: category,
+  });
+
+  // DataSources Views of the system vault holding the managed datasources we want to select data from.
+  const {
```

Comment:
If we move the button within the component, this can be moved as well in the component.

Reply:
I don't think it's clean to put the button in the component called Modal, so I'll create a new one.